modules/stochastic_tools/src/transfers/SerializedSolutionTransfer.C

```diff
@@ -0,0 +1,212 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// StochasticTools includes
+#include "SerializedSolutionTransfer.h"
+#include "NonlinearSystemBase.h"
+#include "Sampler.h"
+
+registerMooseObject("StochasticToolsApp", SerializedSolutionTransfer);
+
+InputParameters
+SerializedSolutionTransfer::validParams()
+{
+  InputParameters params = StochasticToolsTransfer::validParams();
+  params.addClassDescription(
+      "Serializes and transfers solution vectors for given variables from sub-applications.");
+  params.addRequiredParam<std::string>("parallel_storage",
+                                       "The name of the parallel storage reporter.");
+  params.addRequiredParam<std::string>("solution_container",
+                                       "The name of the solution container on the subapp.");
+  params.addRequiredParam<std::vector<VariableName>>(
+      "variables",
+      "The names of the variables which should be serialized and transferred to this application.");
+  params.addParam<bool>(
+      "serialize_on_root",
+      false,
+      "If we only want to gather the solution fields on the root processors of the subapps.");
```

Comment:
why only? we still transfer to the main app parallel storage as well

Reply:
The parallelism on the main app is different. Let's say I have 10 samples on 4 processors and 2 subapps (2 procs for each). On the subapp, we store the petsc vectors split between the local procs. Then we want to transfer it into a storage on the main app. The main app however wants to split the data sample-wise. So we serialize the solutions and transfer them to the main app. This can be done multiple ways:
1. We serialize on the root of the subapp and only those procs will have data on the main app
2. We distribute the samples between the procs of the subapp so we can serialize on all of them

Reply:
I can definitely alter the description!